### ISSUE TITLE
alpaca crash on left click on message

### ISSUE THREAD

**Issue by loulou64490** [Jeffser/Alpaca#298]
**Screenshots**

[Capture vidéo du 2024-09-11 21-06-41.webm](https://github.com/user-attachments/assets/66fe02ab-4b5a-46ab-b4fe-7125a96f5f9a)

**Debugging information**
```
INFO	[main.py | main] Alpaca version: 2.0.0
INFO	[connection_handler.py | start] Starting Alpaca's Ollama instance...
INFO	[connection_handler.py | start] Started Alpaca's Ollama instance
2024/09/11 21:06:42 routes.go:1125: INFO server config env="map[CUDA_VISIBLE_DEVICES: GPU_DEVICE_ORDINAL: HIP_VISIBLE_DEVICES: HSA_OVERRIDE_GFX_VERSION: OLLAMA_DEBUG:true OLLAMA_FLASH_ATTENTION:false OLLAMA_HOST:http://127.0.0.1:11435 OLLAMA_INTEL_GPU:false OLLAMA_KEEP_ALIVE:5m0s OLLAMA_LLM_LIBRARY: OLLAMA_MAX_LOADED_MODELS:0 OLLAMA_MAX_QUEUE:512 OLLAMA_MODELS:/home/loulou/.var/app/com.jeffser.Alpaca/data/.ollama/models OLLAMA_NOHISTORY:false OLLAMA_NOPRUNE:false OLLAMA_NUM_PARALLEL:0 OLLAMA_ORIGINS:[http://localhost https://localhost http://localhost:* https://localhost:* http://127.0.0.1 https://127.0.0.1 http://127.0.0.1:* https://127.0.0.1:* http://0.0.0.0 https://0.0.0.0 http://0.0.0.0:* https://0.0.0.0:* app://* file://* tauri://*] OLLAMA_RUNNERS_DIR: OLLAMA_SCHED_SPREAD:false OLLAMA_TMPDIR: ROCR_VISIBLE_DEVICES:]"
time=2024-09-11T21:06:42.127+02:00 level=INFO source=images.go:753 msg="total blobs: 16"
time=2024-09-11T21:06:42.127+02:00 level=INFO source=images.go:760 msg="total unused blobs removed: 0"
time=2024-09-11T21:06:42.127+02:00 level=INFO source=routes.go:1172 msg="Listening on 127.0.0.1:11435 (version 0.3.9)"
time=2024-09-11T21:06:42.128+02:00 level=INFO source=payload.go:30 msg="extracting embedded files" dir=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama3583294142/runners
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu file=build/linux/x86_64/cpu/bin/libggml.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu file=build/linux/x86_64/cpu/bin/libllama.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu file=build/linux/x86_64/cpu/bin/ollama_llama_server.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx file=build/linux/x86_64/cpu_avx/bin/libggml.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx file=build/linux/x86_64/cpu_avx/bin/libllama.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx file=build/linux/x86_64/cpu_avx/bin/ollama_llama_server.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx2 file=build/linux/x86_64/cpu_avx2/bin/libggml.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx2 file=build/linux/x86_64/cpu_avx2/bin/libllama.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx2 file=build/linux/x86_64/cpu_avx2/bin/ollama_llama_server.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v11 file=build/linux/x86_64/cuda_v11/bin/libggml.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v11 file=build/linux/x86_64/cuda_v11/bin/libllama.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v11 file=build/linux/x86_64/cuda_v11/bin/ollama_llama_server.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v12 file=build/linux/x86_64/cuda_v12/bin/libggml.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v12 file=build/linux/x86_64/cuda_v12/bin/libllama.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v12 file=build/linux/x86_64/cuda_v12/bin/ollama_llama_server.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=rocm_v60102 file=build/linux/x86_64/rocm_v60102/bin/libggml.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=rocm_v60102 file=build/linux/x86_64/rocm_v60102/bin/libllama.so.gz
time=2024-09-11T21:06:42.128+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=rocm_v60102 file=build/linux/x86_64/rocm_v60102/bin/ollama_llama_server.gz
INFO	[connection_handler.py | start] Ollama version: 0.3.9
INFO	[connection_handler.py | request] GET : http://127.0.0.1:11435/api/tags
time=2024-09-11T21:06:51.913+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama3583294142/runners/cpu/ollama_llama_server
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama3583294142/runners/cpu_avx/ollama_llama_server
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama3583294142/runners/cpu_avx2/ollama_llama_server
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama3583294142/runners/cuda_v11/ollama_llama_server
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama3583294142/runners/cuda_v12/ollama_llama_server
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama3583294142/runners/rocm_v60102/ollama_llama_server
time=2024-09-11T21:06:51.914+02:00 level=INFO source=payload.go:44 msg="Dynamic LLM libraries [cpu cpu_avx cpu_avx2 cuda_v11 cuda_v12 rocm_v60102]"
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=payload.go:45 msg="Override detection logic by setting OLLAMA_LLM_LIBRARY"
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=sched.go:105 msg="starting llm scheduler"
time=2024-09-11T21:06:51.914+02:00 level=INFO source=gpu.go:200 msg="looking for compatible GPUs"
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=gpu.go:86 msg="searching for GPU discovery libraries for NVIDIA"
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=gpu.go:468 msg="Searching for GPU library" name=libcuda.so*
time=2024-09-11T21:06:51.914+02:00 level=DEBUG source=gpu.go:491 msg="gpu library search" globs="[/app/lib/ollama/libcuda.so* /app/lib/libcuda.so* /usr/lib/x86_64-linux-gnu/GL/default/lib/libcuda.so* /usr/lib/x86_64-linux-gnu/openh264/extra/libcuda.so* /usr/lib/x86_64-linux-gnu/openh264/extra/libcuda.so* /usr/lib/sdk/llvm15/lib/libcuda.so* /usr/lib/x86_64-linux-gnu/GL/default/lib/libcuda.so* /usr/lib/ollama/libcuda.so* /app/plugins/AMD/lib/ollama/libcuda.so* /usr/local/cuda*/targets/*/lib/libcuda.so* /usr/lib/*-linux-gnu/nvidia/current/libcuda.so* /usr/lib/*-linux-gnu/libcuda.so* /usr/lib/wsl/lib/libcuda.so* /usr/lib/wsl/drivers/*/libcuda.so* /opt/cuda/lib*/libcuda.so* /usr/local/cuda/lib*/libcuda.so* /usr/lib*/libcuda.so* /usr/local/lib*/libcuda.so*]"
time=2024-09-11T21:06:51.915+02:00 level=DEBUG source=gpu.go:525 msg="discovered GPU libraries" paths=[]
time=2024-09-11T21:06:51.915+02:00 level=DEBUG source=gpu.go:468 msg="Searching for GPU library" name=libcudart.so*
time=2024-09-11T21:06:51.915+02:00 level=DEBUG source=gpu.go:491 msg="gpu library search" globs="[/app/lib/ollama/libcudart.so* /app/lib/libcudart.so* /usr/lib/x86_64-linux-gnu/GL/default/lib/libcudart.so* /usr/lib/x86_64-linux-gnu/openh264/extra/libcudart.so* /usr/lib/x86_64-linux-gnu/openh264/extra/libcudart.so* /usr/lib/sdk/llvm15/lib/libcudart.so* /usr/lib/x86_64-linux-gnu/GL/default/lib/libcudart.so* /usr/lib/ollama/libcudart.so* /app/plugins/AMD/lib/ollama/libcudart.so* /home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama3583294142/runners/cuda*/libcudart.so* /usr/local/cuda/lib64/libcudart.so* /usr/lib/x86_64-linux-gnu/nvidia/current/libcudart.so* /usr/lib/x86_64-linux-gnu/libcudart.so* /usr/lib/wsl/lib/libcudart.so* /usr/lib/wsl/drivers/*/libcudart.so* /opt/cuda/lib64/libcudart.so* /usr/local/cuda*/targets/aarch64-linux/lib/libcudart.so* /usr/lib/aarch64-linux-gnu/nvidia/current/libcudart.so* /usr/lib/aarch64-linux-gnu/libcudart.so* /usr/local/cuda/lib*/libcudart.so* /usr/lib*/libcudart.so* /usr/local/lib*/libcudart.so*]"
time=2024-09-11T21:06:51.916+02:00 level=DEBUG source=gpu.go:525 msg="discovered GPU libraries" paths="[/app/lib/ollama/libcudart.so.12.4.99 /app/lib/ollama/libcudart.so.11.3.109]"
cudaSetDevice err: 35
time=2024-09-11T21:06:51.916+02:00 level=DEBUG source=gpu.go:537 msg="Unable to load cudart" library=/app/lib/ollama/libcudart.so.12.4.99 error="your nvidia driver is too old or missing.  If you have a CUDA GPU please upgrade to run ollama"
cudaSetDevice err: 35
time=2024-09-11T21:06:51.917+02:00 level=DEBUG source=gpu.go:537 msg="Unable to load cudart" library=/app/lib/ollama/libcudart.so.11.3.109 error="your nvidia driver is too old or missing.  If you have a CUDA GPU please upgrade to run ollama"
time=2024-09-11T21:06:51.917+02:00 level=DEBUG source=amd_linux.go:371 msg="amdgpu driver not detected /sys/module/amdgpu"
time=2024-09-11T21:06:51.917+02:00 level=INFO source=gpu.go:347 msg="no compatible GPUs were discovered"
time=2024-09-11T21:06:51.917+02:00 level=INFO source=types.go:107 msg="inference compute" id=0 library=cpu variant=avx2 compute="" driver=0.0 name="" total="31.2 GiB" available="26.7 GiB"
[GIN] 2024/09/11 - 21:06:51 | 200 |     526.686µs |       127.0.0.1 | GET      "/api/tags"

(alpaca:2): Gtk-WARNING **: 21:06:52.132: Failed to set text 'html
<div class="center-me">
  <!-- your content here -->
</div>
<tt></tt>' from markup due to error parsing markup: Unknown tag 'div' on line 3 char 1

(alpaca:2): Gtk-WARNING **: 21:06:52.133: Failed to set text 'html
<div class="center-me">
  <!-- your content here -->
</div>
<tt></tt>' from markup due to error parsing markup: Unknown tag 'div' on line 3 char 1

(alpaca:2): Gtk-WARNING **: 21:06:52.134: Failed to set text 'html
<div class="center-me">
  <!-- your content here -->
</div>
<tt></tt>' from markup due to error parsing markup: Unknown tag 'div' on line 3 char 1
```

**Comment by Jeffser** [comment]
That's interesting, it doesn't even have an error message

**Comment by loulou64490** [comment]
when I launch the app after the crash, ollama doesn't work

[Capture vidéo du 2024-09-11 21-27-39.webm](https://github.com/user-attachments/assets/8d09546b-fa31-466a-8978-d32bbe366268)
```
INFO	[main.py | main] Alpaca version: 2.0.0
INFO	[connection_handler.py | start] Starting Alpaca's Ollama instance...
INFO	[connection_handler.py | start] Started Alpaca's Ollama instance
Error: listen tcp 127.0.0.1:11435: bind: address already in use
INFO	[connection_handler.py | start] Ollama version: 0.3.9
INFO	[connection_handler.py | request] GET : http://127.0.0.1:11435/api/tags
ERROR	[model_widget.py | update_local_list] ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
ERROR	[window.py | connection_error] Connection error
INFO	[connection_handler.py | reset] Resetting Alpaca's Ollama instance
INFO	[connection_handler.py | stop] Stopping Alpaca's Ollama instance
INFO	[connection_handler.py | stop] Stopped Alpaca's Ollama instance
INFO	[connection_handler.py | start] Starting Alpaca's Ollama instance...
INFO	[connection_handler.py | start] Started Alpaca's Ollama instance
2024/09/11 21:27:54 routes.go:1125: INFO server config env="map[CUDA_VISIBLE_DEVICES: GPU_DEVICE_ORDINAL: HIP_VISIBLE_DEVICES: HSA_OVERRIDE_GFX_VERSION: OLLAMA_DEBUG:true OLLAMA_FLASH_ATTENTION:false OLLAMA_HOST:http://127.0.0.1:11435 OLLAMA_INTEL_GPU:false OLLAMA_KEEP_ALIVE:5m0s OLLAMA_LLM_LIBRARY: OLLAMA_MAX_LOADED_MODELS:0 OLLAMA_MAX_QUEUE:512 OLLAMA_MODELS:/home/loulou/.var/app/com.jeffser.Alpaca/data/.ollama/models OLLAMA_NOHISTORY:false OLLAMA_NOPRUNE:false OLLAMA_NUM_PARALLEL:0 OLLAMA_ORIGINS:[http://localhost https://localhost http://localhost:* https://localhost:* http://127.0.0.1 https://127.0.0.1 http://127.0.0.1:* https://127.0.0.1:* http://0.0.0.0 https://0.0.0.0 http://0.0.0.0:* https://0.0.0.0:* app://* file://* tauri://*] OLLAMA_RUNNERS_DIR: OLLAMA_SCHED_SPREAD:false OLLAMA_TMPDIR: ROCR_VISIBLE_DEVICES:]"
time=2024-09-11T21:27:54.348+02:00 level=INFO source=images.go:753 msg="total blobs: 16"
time=2024-09-11T21:27:54.349+02:00 level=INFO source=images.go:760 msg="total unused blobs removed: 0"
time=2024-09-11T21:27:54.349+02:00 level=INFO source=routes.go:1172 msg="Listening on 127.0.0.1:11435 (version 0.3.9)"
time=2024-09-11T21:27:54.349+02:00 level=INFO source=payload.go:30 msg="extracting embedded files" dir=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama800023191/runners
time=2024-09-11T21:27:54.349+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu file=build/linux/x86_64/cpu/bin/libggml.so.gz
time=2024-09-11T21:27:54.349+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu file=build/linux/x86_64/cpu/bin/libllama.so.gz
time=2024-09-11T21:27:54.349+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu file=build/linux/x86_64/cpu/bin/ollama_llama_server.gz
time=2024-09-11T21:27:54.349+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx file=build/linux/x86_64/cpu_avx/bin/libggml.so.gz
time=2024-09-11T21:27:54.349+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx file=build/linux/x86_64/cpu_avx/bin/libllama.so.gz
time=2024-09-11T21:27:54.349+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx file=build/linux/x86_64/cpu_avx/bin/ollama_llama_server.gz
time=2024-09-11T21:27:54.349+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx2 file=build/linux/x86_64/cpu_avx2/bin/libggml.so.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx2 file=build/linux/x86_64/cpu_avx2/bin/libllama.so.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cpu_avx2 file=build/linux/x86_64/cpu_avx2/bin/ollama_llama_server.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v11 file=build/linux/x86_64/cuda_v11/bin/libggml.so.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v11 file=build/linux/x86_64/cuda_v11/bin/libllama.so.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v11 file=build/linux/x86_64/cuda_v11/bin/ollama_llama_server.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v12 file=build/linux/x86_64/cuda_v12/bin/libggml.so.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v12 file=build/linux/x86_64/cuda_v12/bin/libllama.so.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=cuda_v12 file=build/linux/x86_64/cuda_v12/bin/ollama_llama_server.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=rocm_v60102 file=build/linux/x86_64/rocm_v60102/bin/libggml.so.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=rocm_v60102 file=build/linux/x86_64/rocm_v60102/bin/libllama.so.gz
time=2024-09-11T21:27:54.350+02:00 level=DEBUG source=payload.go:182 msg=extracting variant=rocm_v60102 file=build/linux/x86_64/rocm_v60102/bin/ollama_llama_server.gz
INFO	[connection_handler.py | start] Ollama version: 0.3.9
INFO	[window.py | show_toast] Une erreur est survenue avec l'instance local d'Ollama, elle a donc été reinitialisée

(alpaca:2): Gtk-WARNING **: 21:27:54.739: Failed to set text 'html
<div class="center-me">
  <!-- your content here -->
</div>
<tt></tt>' from markup due to error parsing markup: Unknown tag 'div' on line 3 char 1

(alpaca:2): Gtk-WARNING **: 21:27:54.741: Failed to set text 'html
<div class="center-me">
  <!-- your content here -->
</div>
<tt></tt>' from markup due to error parsing markup: Unknown tag 'div' on line 3 char 1

(alpaca:2): Gtk-WARNING **: 21:27:54.746: Failed to set text 'html
<div class="center-me">
  <!-- your content here -->
</div>
<tt></tt>' from markup due to error parsing markup: Unknown tag 'div' on line 3 char 1
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama800023191/runners/cpu/ollama_llama_server
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama800023191/runners/cpu_avx/ollama_llama_server
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama800023191/runners/cpu_avx2/ollama_llama_server
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama800023191/runners/cuda_v11/ollama_llama_server
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama800023191/runners/cuda_v12/ollama_llama_server
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=payload.go:71 msg="availableServers : found" file=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama800023191/runners/rocm_v60102/ollama_llama_server
time=2024-09-11T21:28:06.135+02:00 level=INFO source=payload.go:44 msg="Dynamic LLM libraries [cpu cpu_avx cpu_avx2 cuda_v11 cuda_v12 rocm_v60102]"
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=payload.go:45 msg="Override detection logic by setting OLLAMA_LLM_LIBRARY"
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=sched.go:105 msg="starting llm scheduler"
time=2024-09-11T21:28:06.135+02:00 level=INFO source=gpu.go:200 msg="looking for compatible GPUs"
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=gpu.go:86 msg="searching for GPU discovery libraries for NVIDIA"
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=gpu.go:468 msg="Searching for GPU library" name=libcuda.so*
time=2024-09-11T21:28:06.135+02:00 level=DEBUG source=gpu.go:491 msg="gpu library search" globs="[/app/lib/ollama/libcuda.so* /app/lib/libcuda.so* /usr/lib/x86_64-linux-gnu/GL/default/lib/libcuda.so* /usr/lib/x86_64-linux-gnu/openh264/extra/libcuda.so* /usr/lib/x86_64-linux-gnu/openh264/extra/libcuda.so* /usr/lib/sdk/llvm15/lib/libcuda.so* /usr/lib/x86_64-linux-gnu/GL/default/lib/libcuda.so* /usr/lib/ollama/libcuda.so* /app/plugins/AMD/lib/ollama/libcuda.so* /usr/local/cuda*/targets/*/lib/libcuda.so* /usr/lib/*-linux-gnu/nvidia/current/libcuda.so* /usr/lib/*-linux-gnu/libcuda.so* /usr/lib/wsl/lib/libcuda.so* /usr/lib/wsl/drivers/*/libcuda.so* /opt/cuda/lib*/libcuda.so* /usr/local/cuda/lib*/libcuda.so* /usr/lib*/libcuda.so* /usr/local/lib*/libcuda.so*]"
time=2024-09-11T21:28:06.137+02:00 level=DEBUG source=gpu.go:525 msg="discovered GPU libraries" paths=[]
time=2024-09-11T21:28:06.137+02:00 level=DEBUG source=gpu.go:468 msg="Searching for GPU library" name=libcudart.so*
time=2024-09-11T21:28:06.137+02:00 level=DEBUG source=gpu.go:491 msg="gpu library search" globs="[/app/lib/ollama/libcudart.so* /app/lib/libcudart.so* /usr/lib/x86_64-linux-gnu/GL/default/lib/libcudart.so* /usr/lib/x86_64-linux-gnu/openh264/extra/libcudart.so* /usr/lib/x86_64-linux-gnu/openh264/extra/libcudart.so* /usr/lib/sdk/llvm15/lib/libcudart.so* /usr/lib/x86_64-linux-gnu/GL/default/lib/libcudart.so* /usr/lib/ollama/libcudart.so* /app/plugins/AMD/lib/ollama/libcudart.so* /home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama800023191/runners/cuda*/libcudart.so* /usr/local/cuda/lib64/libcudart.so* /usr/lib/x86_64-linux-gnu/nvidia/current/libcudart.so* /usr/lib/x86_64-linux-gnu/libcudart.so* /usr/lib/wsl/lib/libcudart.so* /usr/lib/wsl/drivers/*/libcudart.so* /opt/cuda/lib64/libcudart.so* /usr/local/cuda*/targets/aarch64-linux/lib/libcudart.so* /usr/lib/aarch64-linux-gnu/nvidia/current/libcudart.so* /usr/lib/aarch64-linux-gnu/libcudart.so* /usr/local/cuda/lib*/libcudart.so* /usr/lib*/libcudart.so* /usr/local/lib*/libcudart.so*]"
time=2024-09-11T21:28:06.138+02:00 level=DEBUG source=gpu.go:525 msg="discovered GPU libraries" paths="[/app/lib/ollama/libcudart.so.12.4.99 /app/lib/ollama/libcudart.so.11.3.109]"
cudaSetDevice err: 35
time=2024-09-11T21:28:06.138+02:00 level=DEBUG source=gpu.go:537 msg="Unable to load cudart" library=/app/lib/ollama/libcudart.so.12.4.99 error="your nvidia driver is too old or missing.  If you have a CUDA GPU please upgrade to run ollama"
cudaSetDevice err: 35
time=2024-09-11T21:28:06.138+02:00 level=DEBUG source=gpu.go:537 msg="Unable to load cudart" library=/app/lib/ollama/libcudart.so.11.3.109 error="your nvidia driver is too old or missing.  If you have a CUDA GPU please upgrade to run ollama"
time=2024-09-11T21:28:06.138+02:00 level=DEBUG source=amd_linux.go:371 msg="amdgpu driver not detected /sys/module/amdgpu"
time=2024-09-11T21:28:06.138+02:00 level=INFO source=gpu.go:347 msg="no compatible GPUs were discovered"
time=2024-09-11T21:28:06.138+02:00 level=INFO source=types.go:107 msg="inference compute" id=0 library=cpu variant=avx2 compute="" driver=0.0 name="" total="31.2 GiB" available="26.7 GiB"
INFO	[window.py | closing_app] Closing app...
INFO	[connection_handler.py | stop] Stopping Alpaca's Ollama instance
time=2024-09-11T21:28:15.099+02:00 level=DEBUG source=sched.go:119 msg="shutting down scheduler pending loop"
time=2024-09-11T21:28:15.099+02:00 level=DEBUG source=sched.go:318 msg="shutting down scheduler completed loop"
time=2024-09-11T21:28:15.099+02:00 level=DEBUG source=assets.go:115 msg="cleaning up" dir=/home/loulou/.var/app/com.jeffser.Alpaca/cache/tmp/ollama/ollama800023191
INFO	[connection_handler.py | stop] Stopped Alpaca's Ollama instance
```

**Comment by olumolu** [comment]
Which os are you using

**Comment by loulou64490** [comment]
Fedora 39 workstation

**Comment by Jeffser** [comment]
Hi, I figured out what was the problem, I made some code that deselected all the text when the user focuses a different element so it doesn't get stuck selected, the context menu that appears when the user right clicks the text requires it to be selectable and since it both opens the context menu and makes it not selectable at the same time, it crashes the app.

By inverting the condition of focus and adding it to the GTK thread it makes it work!

tldr: Code bad, now good

https://github.com/Jeffser/Alpaca/commit/f8d18afd13ae739ba50bbd22da1682d22763cef1
